lib/backend/backend.go

```diff
@@ -400,3 +400,31 @@ func NewLease(item Item) *Lease {
 		Revision: item.Revision,
 	}
 }
+
+// PutBatcher is an optional interface that backends can implement
+// to support batched PutBatch operations for improved performance when writing
+// multiple items at once.
+type PutBatcher interface {
+	// PutBatch puts multiple items into the backend in a single batch operation.
+	// Returns a revision item for each item in the same order.
+	PutBatch(context.Context, []Item) ([]string, error)
```

Comment:
Do we want to put in some limitation to how many writes can happen in a single batch, or should it be the responsibility of the backend implementation to figure out the best way to split this up in internal operations for the specific storage service?

Reply:
> or should it be the responsibility of the backend implementation to figure out the best way to split this up in internal operations for the specific storage service?

This shoudl be up to the backend since backend are aware about thier limitation